Modules/_io/fileio.c

```diff
@@ -455,7 +462,9 @@ _io_FileIO___init___impl(fileio *self, PyObject *nameobj, const char *mode,
 #endif
     }
 
-    PyMem_Free(self->stat_atopen);
+    if (self->stat_atopen != NULL) {
```

Comment:
PyMem_Free(NULL) is well defined: it does nothing, so you might omit the `if (stat_atopen != NULL)` test.